matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/auth/version/HomeServerVersion.kt

```diff
@@ -59,3 +59,5 @@ internal data class HomeServerVersion(
         val v1_3_0 = HomeServerVersion(major = 1, minor = 3, patch = 0)
     }
 }
+
+private fun List<String>.getOptional(index: Int, default: String) = getOrNull(index)?.ifEmpty { default } ?: default
```

Comment:
`ifEmpty` is maybe not necessary. I do not think it can be empty, since we have `\d+`. Maybe add a test with "v1.4.", expecting `null`.

Reply:
the optional block returns an empty string, with `ifEmpty` removed

```
java.lang.NumberFormatException: For input string: ""
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.base/java.lang.Integer.parseInt(Integer.java:662)
	at java.base/java.lang.Integer.parseInt(Integer.java:770)
	at org.matrix.android.sdk.internal.auth.version.HomeServerVersion$Companion.parse$matrix_sdk_android_debug(HomeServerVersion.kt:48)
```

have updated the test cases to include trailing `.` https://github.com/vector-im/element-android/pull/6214/commits/4501c7cf45ebd37bd5fdc190a0ea5cff2bc68341

Reply:
I see, thanks. I would avoid creating this extension and would have written at line 48:
```kotlin
patch = result.groupValues.getOrNull(3)?.ensureNotEmpty()?.toInt() ?: 0
```

Since, this is not necessary to do `"0".toInt()`.

The extension:
```kotlin
fun String.ensureNotEmpty() = takeIf { it.isNotEmpty() }
```
could be added to the core <del>module</del> package. I am surprised we do not have create it yet, we have 76 occurrences of `.takeIf { it.isNotEmpty() }` in the codebase :)

This is obviously not a blocker :).